src/guest_memory.rs

```diff
@@ -295,8 +295,8 @@ pub trait GuestMemory {
         let mut total = 0;
         while let Some(region) = self.find_region(cur) {
             let start = region.to_region_addr(cur).unwrap();
-            let cap = region.len() as usize;
-            let len = std::cmp::min(cap, count - total);
+            let size = region.len() as usize - start.raw_value() as usize;
```

Comment:
Sorry, at a second glance found it's unsafe to cast region.len() and start.raw_value() as usize.
usize is 32bit with a 32bit VMM but region.len() and start.raw_value() are both u64.

Reply:
Yes, this must be `GuestUsize`.

Reply:
It seems this is not the only case in guest_memory.rs regarding the potential truncate issue, I suggest to have a separate PR opened to handle such issue.

Reply:
Still, this should use `GuestUsize`.